flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/KafkaWriter.java

```diff
@@ -139,6 +147,7 @@ public void write(IN element, Context context) throws IOException {
                 recordSerializer.serialize(element, kafkaSinkContext, context.timestamp());
         pendingRecords.incrementAndGet();
         currentProducer.send(record, deliveryCallback);
+        MetricUtil.sync(byteOutMetric, numBytesOutCounter);
```

Comment:
I just saw that `metricValue` is under lock, so this is probably too expensive for every record. How about adding a timerservice callback every second that syncs?
Or we could use `Partitioner#newBatch` to sync.

Reply:
I registered a timer to update the metrics every 500ms, let me know if this is a reasonable time.